backend/Bindings/Edna.Bindings.LtiAdvantage/Services/NrpsClient.cs

```diff
@@ -70,18 +71,27 @@ public async Task<IEnumerable<Member>> Get(string clientId, string tokenUrl, str
             httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(Constants.MediaTypes.MembershipContainer));
 
             _logger.LogInformation("Getting members.");
-            using var response = await httpClient.GetAsync(membershipUrl);
+            List<Member> members = new List<Member>();
 
-            if (!response.IsSuccessStatusCode)
+            do

```

Comment:
Quick Question on the `do..while`, it looks like we are proactively fetching all the results and sending them over to the client here. I was wondering if this could be done lazily to allow the client to call the next `membershipurl` as and when needed. Does that mean changing a lot of functionality on the client - such as `filter`, `sortBy` and `search`? If not, probably we can try to use that approach to help reduce user bandwidth for these cases 🤔 ...

Reply:
Maitree's response:
1. It will change client functionality
2. Twice in a launch (me call and get all users call)

Reply:
Makes sense. Thanks for explaining. Is it possible to get some idea about how much latency increase this would cause for a target class of let's say 100 students. I randomly picked up 100, the actual number can be picked up from our customers (@leestott - any thoughts on what would be a good number to benchmark against?)